src/outputProcessor/NestedOutputProcessorDecorator.js

```diff
@@ -21,7 +21,7 @@ module.exports = class NestedOutputProcessorDecorator {
   write(allMessages) {
     Object.entries(allMessages).forEach(([spec, messages]) => {
       const relativeSpec = path.relative(this.specRoot, spec);
-      const outPath = path.join(this.root, relativeSpec.replace(/\..+$/, `.${this.ext}`));
+      const outPath = path.join(this.root, relativeSpec.replace(new RegExp(path.extname(relativeSpec) + '$'), `.${this.ext}`));
```

Comment:
Yeah this regex was not quite correct. But the better way to do this is with the following regex `/\.[^\.]+/`. Can you also modify or add a spec file to the test that has such name?

Reply:
@archfz Thanks!

I thought about using this regex `/\.[^\.]+$/` but I think my version is a bit more explicit, I want to find the current extension at the end replace it with the requested extension - I don't care about the location of the dots or anything, just the extension itself.

If you feel that yours is the way to go I'll change it :)

Re: Tests
I sure will :)